catalogue_pipeline/ingestor/src/main/scala/uk/ac/wellcome/platform/ingestor/services/WorkIndexer.scala

```diff
@@ -33,12 +31,11 @@ class WorkIndexer @Inject()(
       "ingestor-index-work",
       () => {
         Future
-          .fromTry(Try {
-            decode[Work](document) match {
-              case Right(work) => work
-              case Left(error) => throw SQSReaderGracefulException(error)
+          .fromTry(
+            fromJson[Work](document).recover {
+              case error: Throwable => throw GracefulFailureException(error)
```

Comment:
Note to self: this clashes with #1427, and that will need to be rebased atop this later.